ItemSearchPlugin/Filters/ItemUICategorySearchFilter.cs

```diff
@@ -33,15 +33,16 @@ public override bool HasChanged {
         private readonly Vector2 popupSize = new Vector2(-1, 120);
 
         public ItemUICategorySearchFilter(ItemSearchPluginConfig config, IDataManager data) : base(config) {
-            uiCategories = new List<ItemUICategory> {null};
-            uiCategories.AddRange(data.GetExcelSheet<ItemUICategory>().ToList().Where(x => !string.IsNullOrEmpty(x.Name)).OrderBy(x => x.Name.ToString()));
+            uiCategories = [null];
+            uiCategories.AddRange(data.GetExcelSheet<ItemUICategory>().ToList().Where(x => !string.IsNullOrEmpty(x.Name.ToString())).OrderBy(x => x.Name.ToString()).Select((x) => (ItemUICategory?)x));
             string nullName = Loc.Localize("ItemUiCategorySearchFilterAll", "All");
-            uiCategoriesArray = uiCategories.Select(x => x == null ? nullName : x.Name.ToString().Replace("\u0002\u001F\u0001\u0003", "-")).ToArray();
+            uiCategoriesArray = uiCategories.Select(x => x == null ? nullName : x.Value.Name.ToString().Replace("\u0002\u001F\u0001\u0003", "-")).ToArray();
```

Comment:
.ToString causes an error in the German translation here
![image](https://github.com/user-attachments/assets/6652dec5-db7f-4d0f-a0bd-c5bcd2877a18)

.ExtractText is a little better, but also not correct
![grafik](https://github.com/user-attachments/assets/c522e29b-9838-4aa6-b788-6a8ea24dd105)

If I have understood it correctly, .ExtractText should generally be used here rather than .ToString

Reply:
ExtractText() keeps a few special characters around, including a soft-hyphen (\u00AD) which is used in the German localization for word breaking. ImGui always prints it as a dash, and it also breaks text matching.

The calls could be wrapped by a method like Dalamud.Game.Text.Sanitizer.Sanitizer.Sanitize(), which seems to be doing the same type of sanitation for soft-hyphens and some other characters such as the French œ character.

Reply:
`Dalamud.Game.Text.Sanitizer.Sanitizer.Sanitize()` doesn't work for this unfortunately, but I got it to work with `x.Value.Name.ToDalamudString().TextValue` instead of `x.Value.Name.ToString()` or `x.Value.Name.ExtractText()`